lib/shopify-cli/contextual_command.rb

```diff
@@ -0,0 +1,28 @@
+require 'shopify_cli'
+
+module ShopifyCli
+  class ContextualCommand < ShopifyCli::Command
+    class << self
+      def needs_contextual_resolution?
+        true
+      end
+
+      def available_in_contexts(command, context_types)
+        project_context = Project.current_context
+        unregister_for_context(command) unless context_types.include?(project_context)
+      end
+
+      def override_in_contexts(command, context_types, path)
```

Comment:
Having the ability to override commands at run time seems more complicated than just having two different command classes and having the registry know how to call the right one based on context.